### PR TITLE
More constness for http_inspector

### PR DESCRIPTION
**Pull request by rgs1**
Bumped into these non consts while fetching  #7559 for our repo.

Signed-off-by: Raul Gutierrez Segales <rgs@pinterest.com>

### CODE REVIEW

**Comment by PiotrSikora** [comment]
cc @crazyxy

File: test/extensions/filters/listener/http_inspector/http_inspector_test.cc [hunk 1]
```diff
@@ -42,6 +42,7 @@ class HttpInspectorTest : public testing::Test {
     EXPECT_CALL(socket_, detectedTransportProtocol()).WillRepeatedly(Return("raw_buffer"));
     EXPECT_CALL(cb_, dispatcher()).WillRepeatedly(ReturnRef(dispatcher_));
     EXPECT_CALL(socket_, ioHandle()).WillRepeatedly(ReturnRef(*io_handle_));
+    EXPECT_CALL(testing::Const(socket_), ioHandle()).WillRepeatedly(ReturnRef(*io_handle_));
```
Comment (PiotrSikora) [inline test/extensions/filters/listener/http_inspector/http_inspector_test.cc]: Do we need both const and non-const variants?

Comment (rgs1) [reply test/extensions/filters/listener/http_inspector/http_inspector_test.cc]: updated so that it isn't needed anymore.

**Comment by PiotrSikora** [comment]
@rgs1 what compiler / flags are you using that triggered this? I think it would be useful to incorporate those into CI.

**Comment by rgs1** [comment]
Mostly my eyes :-)

But @derekarg has been making our clang-tidy rules more and more aggressive
(both upstream & locally).

**Comment by rgs1** [comment]
@PiotrSikora to clarify, we tend to do a more careful read of PRs we'll probably use (or that affect us directly), that's how this was caught.

It would indeed be nice to have tooling catch this. @derekargueta maybe has ideas for this.

**Pull request merged**